esp-wifi/src/ble/os_adapter_esp32c2.rs

```diff
@@ -123,3 +128,22 @@ pub(super) unsafe extern "C" fn ble_ll_random_override() -> u32 {
         VALUE
     }
 }
+
+pub(super) fn deinit() {
+    unsafe {
+        // HCI deinit
+        npl::r_ble_hci_trans_cfg_hs(None, core::ptr::null(), None, core::ptr::null());
+
+        let res = npl::ble_controller_deinit();
+
+        if res != 0 {
+            panic!("ble_controller_deinit returned {}", res);
+        }
+
+        bt_periph_module_disable();
+
+        npl::esp_unregister_npl_funcs();
+        npl::esp_unregister_ext_funcs();
+        crate::common_adapter::chip_specific::phy_disable();
+    }
+}
```

Comment:
This is mostly the same for all NPL chips? Maybe better to place it in npl.rs (especially since it's almost calling functions there)

One exception might be `crate::common_adapter::chip_specific::phy_disable();` - but why only for C2 (for NPLs) and shouldn't that be done in deinit after everything else is de-inited?